pkg/io/providers/aws.go

```diff
@@ -57,9 +60,24 @@ func getS3Bucket(ctx context.Context, creds []byte, bucketName string) (*blob.Bu
 		return nil, fmt.Errorf("error loading AWS SDK config: %w", err)
 	}
 
-	client := s3.NewFromConfig(cfg, func(o *s3.Options) {
-		o.UsePathStyle = s3Creds.S3ForcePathStyle
-	})
+	s3Opts := []func(o *s3.Options){
+		func(o *s3.Options) {
+			o.UsePathStyle = s3Creds.S3ForcePathStyle
+			o.HTTPClient = awshttp.NewBuildableClient().WithTransportOptions(func(t *http.Transport) {
+				if t.TLSClientConfig == nil {
+					t.TLSClientConfig = &tls.Config{}
+				}
+				t.TLSClientConfig.InsecureSkipVerify = s3Creds.Insecure
```

Comment:
Why are we unconditionally doing this?

Reply:
The calling code in the aws sdk is the following:
```go
func resolveHTTPClient(o *Options) {
	var buildable *awshttp.BuildableClient

	if o.HTTPClient != nil {
		var ok bool
		buildable, ok = o.HTTPClient.(*awshttp.BuildableClient)
		if !ok {
			return
		}
	} else {
		buildable = awshttp.NewBuildableClient()
	}

	modeConfig, err := defaults.GetModeConfiguration(o.resolvedDefaultsMode)
	if err == nil {
		buildable = buildable.WithDialerOptions(func(dialer *net.Dialer) {
			if dialerTimeout, ok := modeConfig.GetConnectTimeout(); ok {
				dialer.Timeout = dialerTimeout
			}
		})

		buildable = buildable.WithTransportOptions(func(transport *http.Transport) {
			if tlsHandshakeTimeout, ok := modeConfig.GetTLSNegotiationTimeout(); ok {
				transport.TLSHandshakeTimeout = tlsHandshakeTimeout
			}
		})
	}

	o.HTTPClient = buildable
}
```
Since `s3Creds.Insecure `defaults to false, it does change the behavior only if it's set to true, so no need to check first if its true.

Reply:
I see. `*bool` might have been nice for "no opinion" but this seems fine. Do you think we can write a unit test for `s3Creds -> *s3.Options` that builds the mutation chain and executes it? Or - how to ensure that all the fields in the input options are actually translated to the s3 SDK?

Reply:
I've implemented tests for creating a S3 client